semgrep/semgrep/core_runner.py

```diff
@@ -613,90 +613,86 @@ def _run_rules_direct_to_semgrep_core(
         Set[Path],
         ProfilingData,
     ]:
-        from itertools import chain
-        from collections import defaultdict
-
         logger.debug(f"Passing whole rules directly to semgrep_core")
 
-        outputs: Dict[Rule, List[RuleMatch]] = defaultdict(list)
+        outputs: Dict[Rule, List[RuleMatch]] = collections.defaultdict(list)
         errors: List[SemgrepError] = []
         all_targets: Set[Path] = set()
         profiling_data: ProfilingData = ProfilingData()
         # cf. for bar_format: https://tqdm.github.io/docs/tqdm/
         with tempfile.TemporaryDirectory() as semgrep_core_ast_cache_dir:
-            for rule, language in tuple(
-                chain(
-                    *(
-                        [(rule, language) for language in rule.languages]
-                        for rule in rules
-                    )
-                )
+            for rule in progress_bar(
+                rules, bar_format="{l_bar}{bar}|{n_fmt}/{total_fmt}"
             ):
```

Comment:
Changing from chain in for loop to double for loop is the only relevant diff in this PR. The rest are indentation changes.

Reply:
github needs to improve his diff algorithm to show those just-change-of-indentation lines in a different color.